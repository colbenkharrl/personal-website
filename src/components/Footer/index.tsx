'use client';

import Link from 'next/link';

import { ContainerInner, ContainerOuter } from '@/components/Container';
import React, { useCallback, useState } from 'react';
import { SnackBar } from '@/components/SnackBar';

function NavLink({
  href,
  children,
}: {
  href: string;
  children: React.ReactNode;
}) {
  return (
    <Link
      href={href}
      className="transition hover:text-yellow-500 dark:hover:text-yellow-400"
    >
      {children}
    </Link>
  );
}

export function Footer() {
  const [isSnackBarShowing, setSnackBarShowing] = useState(true);

  const handleCloseSnackBar = useCallback(() => {
    setSnackBarShowing(false);
  }, []);

  return (
    <footer
      className={`mt-32 flex-none ${isSnackBarShowing ? 'pb-20' : ''}`}
      data-testid="Footer"
    >
      <ContainerOuter>
        <div className="border-t border-zinc-100 pb-16 pt-10 dark:border-zinc-700/40">
          <ContainerInner>
            <div className="flex flex-col items-center justify-between gap-6 sm:flex-row">
              <div className="flex flex-wrap justify-center gap-x-6 gap-y-1 text-sm font-medium text-zinc-800 dark:text-zinc-200">
                <NavLink href="/">Home</NavLink>
                <NavLink href="/about">About</NavLink>
                <NavLink href="/portfolio">Portfolio</NavLink>
                <NavLink href="/blog">Blog</NavLink>
              </div>
              <p className="text-sm text-zinc-400 dark:text-zinc-500">
                &copy; {new Date().getFullYear()} Matt Kharrl. All rights
                reserved.
              </p>
            </div>
          </ContainerInner>
        </div>
      </ContainerOuter>
      <SnackBar onClose={handleCloseSnackBar} closeMs={10000}>
        <Link href="https://github.com/colbenkharrl/personal-website">
          <strong className="font-semibold">
            This site is designed and developed by me.
          </strong>{' '}
          View the the source on GitHub. <span aria-hidden="true">&rarr;</span>
        </Link>
      </SnackBar>
    </footer>
  );
}
