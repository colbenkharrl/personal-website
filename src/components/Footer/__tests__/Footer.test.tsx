import {
  act,
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react';
import * as stories from '../__stories__/Footer.stories';
import { composeStories } from '@storybook/react';

const { Basic } = composeStories(stories);

jest.useFakeTimers();

it('Footer matches snapshot (regression test)', () => {
  render(<Basic />);
  expect(screen.getByTestId('Footer')).toMatchSnapshot(
    'Full component snapshot.',
  );
});

it('should close snackbar', async () => {
  render(<Basic />);

  fireEvent.click(screen.getByTestId('SnackBar-close-icon'));

  act(() => {
    jest.advanceTimersByTime(100000);
  });

  await waitFor(() =>
    expect(screen.queryByTestId('SnackBar-close-icon')).toBeNull(),
  );
});
