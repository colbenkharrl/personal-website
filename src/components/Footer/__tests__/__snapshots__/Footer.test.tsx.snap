// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot (regression test): Full component snapshot. 1`] = `
<footer
  class="mt-32 flex-none pb-20"
  data-testid="Footer"
>
  <div
    class="sm:px-8"
  >
    <div
      class="mx-auto w-full max-w-7xl lg:px-8"
    >
      <div
        class="border-t border-zinc-100 pb-16 pt-10 dark:border-zinc-700/40"
      >
        <div
          class="relative px-4 sm:px-8 lg:px-12"
        >
          <div
            class="mx-auto max-w-2xl lg:max-w-5xl"
          >
            <div
              class="flex flex-col items-center justify-between gap-6 sm:flex-row"
            >
              <div
                class="flex flex-wrap justify-center gap-x-6 gap-y-1 text-sm font-medium text-zinc-800 dark:text-zinc-200"
              >
                <a
                  class="transition hover:text-amber-500 dark:hover:text-amber-400"
                  href="/"
                >
                  Home
                </a>
                <a
                  class="transition hover:text-amber-500 dark:hover:text-amber-400"
                  href="/about"
                >
                  About
                </a>
                <a
                  class="transition hover:text-amber-500 dark:hover:text-amber-400"
                  href="/portfolio"
                >
                  Portfolio
                </a>
                <a
                  class="transition hover:text-amber-500 dark:hover:text-amber-400"
                  href="/blog"
                >
                  Blog
                </a>
              </div>
              <p
                class="text-sm text-zinc-400 dark:text-zinc-500"
              >
                © 
                2025
                 Matt Kharrl. All rights reserved.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="fixed inset-x-0 bottom-0  z-50"
    data-testid="SnackBar"
  >
    <div
      class="opacity-100 flex items-center gap-x-6 bg-white bg-opacity-90 px-6 py-2.5 outline outline-2 outline-amber-500 transition-opacity duration-500 ease-out dark:bg-zinc-900 dark:bg-opacity-90 dark:outline-amber-400 sm:px-3.5 sm:before:flex-1 sm:after:flex-1 "
    >
      <p
        class="z-50 text-sm leading-6 text-zinc-800 dark:text-zinc-100"
      >
        <a
          href="https://github.com/colbenkharrl/personal-website"
        >
          <strong
            class="font-semibold"
          >
            This site is designed and developed by me.
          </strong>
           
          View the the source on GitHub. 
          <span
            aria-hidden="true"
          >
            →
          </span>
        </a>
      </p>
      <button
        class="z-50 -m-1.5 flex-none  p-1.5"
        data-testid="SnackBar-close-icon"
        type="button"
      >
        <span
          class="sr-only"
        >
          Dismiss
        </span>
        <svg
          aria-hidden="true"
          class="h-5 w-5 text-zinc-800 dark:text-zinc-100"
          fill="currentColor"
          viewBox="0 0 20 20"
        >
          <path
            d="M6.28 5.22a.75.75 0 00-1.06 1.06L8.94 10l-3.72 3.72a.75.75 0 101.06 1.06L10 11.06l3.72 3.72a.75.75 0 101.06-1.06L11.06 10l3.72-3.72a.75.75 0 00-1.06-1.06L10 8.94 6.28 5.22z"
          />
        </svg>
      </button>
    </div>
  </div>
</footer>
`;
